lldb/source/Plugins/LanguageRuntime/CPlusPlus/CPPLanguageRuntime.cpp

```diff
@@ -41,7 +41,10 @@ CPPLanguageRuntime::CPPLanguageRuntime(Process *process)
     : LanguageRuntime(process) {}
 
 bool CPPLanguageRuntime::IsAllowedRuntimeValue(ConstString name) {
-  return name == g_this;
+  // FIXME: use a list when the list grows more.
+  return name == g_this ||
+         name == ConstString("__promise") ||
+         name == ConstString("__coro_frame");
```

Comment:
I am not sure if I would expose the `__coro_frame` variable. Maybe I would only expose the `__promise` variable by default. The `__coro_frame` is usually out-of-sync and does not hold the up-to-date coroutine state while the coroutine is in flight

Reply:
Any chance we can get the `Type` from the `Variable` in `CommandObjectFrameVariable` and can re-use the `IsCoroutineFrameType` check?

Reply:
I removed this change from this pull request now (will sent as a followup). Let's focus on addressing the ignore-artificial-fields issue in `DWARFASTParserClang.cpp`.